molecule/src/structure_checker2.cpp

```diff
@@ -0,0 +1,651 @@
+                                           "Structure contains atoms with unusuall valence",
+                                           "Structure contains query features, so valency could not be checked",
+                                           "Structure contains RGroup components, so valency could not be checked",
+                                           "IGNORE_BAD_VALENCE flag is active",
+                                           "Structure contains radicals",
+                                           "Structure contains pseudoatoms, so radicals could not be checked",
+                                           "Structure contains pseudoatoms",
+                                           "Structure contains wrong chiral flag",
+                                           "Structure contains incorrect stereochemistry",
+                                           "Structure contains stereocenters defined by 3D coordinates",
+                                           "Structure contains stereocenters with undefined stereo configuration",
+                                           "CHECK_MSG_IGNORE_STEREO_ERROR",
+                                           "Structure contains query features",
+                                           "Structure contains query features for atoms",
+                                           "Structure contains query features for bonds",
+                                           "CHECK_MSG_IGNORE_QUERY_FEATURE",
+                                           "Structure contains overlapping atoms",
+                                           "Structure contains overlapping bonds.",
+                                           "Structure contains R-groups",
+                                           "Structure contains S-groups",
+                                           "Structure contains SCSR templates",
+                                           "Structure has non-zero charge",
+                                           "Structure contains charged fragments (possible salt)",
+                                           "Input structure is empty",
+                                           "Structure contains ambiguous hydrogens",
+                                           "Structure contains query features, so ambiguous H could not be checked",
+                                           "Structure contains 3D coordinates",
+                                           "Structure has no atoms coordinates"
+                                           "Reaction component check result",
+                                           "Not implemented yet: check chirality",
+                                           "Not implemented yet: check charge",
+                                           "Not implemented yet: check salt",
+                                           "Structure supports only Molfile V3000"
```

Comment:
Not sure why do we need this array - we can instead write these string directly in methods where we address them via enums. Do you think code looks more clear in this approach?

Reply:
this array has specific order for the CheckMessageCode which can be difficult to support. Instead it is worth replace the message_list to message_map<CheckMessageCode , string>